File: src/unitysvc_data/examples/llm/code-example-streaming-openai/code-example-v1.py
import os

from openai import OpenAI

client = OpenAI(
    base_url=os.environ["SERVICE_BASE_URL"],
    api_key=os.environ["UNITYSVC_API_KEY"],
)

stream = client.chat.completions.create(
    model=os.environ["MODEL"],
    messages=[
        {"role": "system", "content": "You are a helpful assistant."},
        {"role": "user", "content": "Say this is a test"},
    ],
    stream=True,
)

for chunk in stream:
    delta = chunk.choices[0].delta.content
    if delta:
        print(delta, end="", flush=True)
print()

File: src/unitysvc_data/examples/llm/code-example-streaming-openai-javascript/code-example-v1.js
const { OpenAI } = require("openai");

const client = new OpenAI({
  baseURL: process.env.SERVICE_BASE_URL,
  apiKey: process.env.UNITYSVC_API_KEY,
});

async function main() {
  const stream = await client.chat.completions.create({
    model: process.env.MODEL,
    messages: [
      { role: "system", content: "You are a helpful assistant." },
      { role: "user", content: "Say this is a test" },
    ],
    stream: true,
  });
  for await (const chunk of stream) {
    const delta = chunk.choices[0]?.delta?.content;
    if (delta) {
      process.stdout.write(delta);
    }
  }
  process.stdout.write("\n");
}

main().catch((err) => {
  console.error(err.message);
  process.exit(1);
});
